module/canvas/layers/tokens.mjs

```diff
@@ -0,0 +1,96 @@
+export default class TokenLayer5e extends foundry.canvas.layers.TokenLayer {
+  /**
+   * Determine whether the provided grid space is being occupied by a token which should block the provided token
+   * @param {GridOffset3D | TokenFindMovementPathWaypoint} gridSpace  The grid space to check
+   * @param {Token5e} token                                           The token being moved
+   * @returns {boolean} Whether the moving token should be blocked
+   */
+  isOccupiedGridSpaceBlocking(gridSpace, token) {
+    const found = this._getRelevantOccupyingTokens(gridSpace, token);
+    const tokenSize = CONFIG.DND5E.actorSizes[token.actor?.system.traits.size]?.numerical ?? 2;
+    const modernRules = game.settings.get("dnd5e", "rulesVersion") === "modern";
+    const halflingNimbleness = token.actor?.getFlag("dnd5e", "halflingNimbleness");
+    return found.some(t => {
+      // Friendly tokens never block movement
+      if ( token.document.disposition === t.document.disposition ) return false;
```

Comment:
I suppose there's no better way to handle this at the moment and we have to assume all hostile tokens are on the same team and all neutral tokens are on the same team.